shogun-lib/src/main/java/de/terrestris/shogun/lib/controller/BaseFileController.java

```diff
@@ -97,31 +107,36 @@
 
             if (entity.isPresent()) {
                 S file = entity.get();
-
-                LOG.info("Successfully got file with UUID {}", fileUuid);
-
                 final HttpHeaders responseHeaders = new HttpHeaders();
                 responseHeaders.setContentType(MediaType.parseMediaType(file.getFileType()));
                 responseHeaders.setContentDisposition(ContentDisposition.parse(
                     String.format("inline; filename=\"%s\"", file.getFileName())));
 
-                return new ResponseEntity<>(file.getFile(), responseHeaders, HttpStatus.OK);
-            } else {
-                LOG.error("Could not find entity of type {} with UUID {}",
-                    getGenericClassName(), fileUuid);
-
-                throw new ResponseStatusException(
-                    HttpStatus.NOT_FOUND,
-                    messageSource.getMessage(
-                        "BaseController.NOT_FOUND",
-                        null,
-                        LocaleContextHolder.getLocale()
-                    )
-                );
+                LOG.trace("Successfully got file with UUID {}", fileUuid);
+                if (file.getPath() == null) {
+                    LOG.trace("… load file from database");
+                    return new ResponseEntity<>(file.getFile(), responseHeaders, HttpStatus.OK);
+                }
+
+                java.io.File dataFile = new java.io.File(uploadBasePath + "/" + file.getPath());
+                if (dataFile.exists()) {
+                    LOG.trace("… load file from disk");
+                    byte[] fileByteArray = FileUtils.readFileToByteArray(dataFile);
+                    return new ResponseEntity<>(fileByteArray, responseHeaders, HttpStatus.OK);
+                }
             }
+
```

Comment:
I would also prefer to move the above code to the service, but this is not blocking.

Reply:
0f92b05